packages/react-core/src/elements/types.ts

```diff
@@ -15,11 +17,40 @@ export type BaseElement<T = {}, K = {}> = React.ForwardRefExoticComponent<
   React.PropsWithoutRef<T> & React.RefAttributes<K>
 >;
 
+type ListItemElementSubType = 'Menu' | 'Nav' | 'List';
+type ListItemElementDisplayName = `${ListItemElementSubType}Item`;
```

Comment:
Does ListItemElement need sub types? Essentially all of these end up being an `<li>` I think. A menu item (role="menuitem") would end up going on a Button that is a child of role="menu".

Reply:
Was thinking more along the lines of differentiation between use cases but am good with removing the sub types, thoughts?

Reply:
Yeah I think I'm good with removing the ListItemElement subtypes; the ones for ListElement and TableElement make sense.